lib/jnpr/junos/factcache.py

```diff
@@ -192,7 +193,7 @@ def __len__(self):
         """
         return len(self._callbacks)
 
-    def __repr__(self):
+    def __str__(self):
```

Comment:
Yes, the string from `repr()` can not yet be passed to `eval()` successfully, but that's because of the string representation of the `jnpr.junos.version_info` objects. We can fix that when I address #651. If we keep the function the `__repr__()` function as it was, and add your new function as `__str__()`, then we most closely mimic the previous behavior when `dev.facts` was a true `dict`.